src/native/minipal/thread.h

```diff
@@ -0,0 +1,110 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#ifndef HAVE_MINIPAL_THREAD_H
+#define HAVE_MINIPAL_THREAD_H
+
+#include <pthread.h>
+#include <stdint.h>
+#include <string.h>
+
+#if defined(__linux__)
+#include <unistd.h>
+#include <sys/syscall.h>
+#elif defined(__FreeBSD__)
+#include <pthread_np.h>
+#elif defined(__NetBSD__)
+#include <lwp.h>
+#elif defined(__HAIKU__)
+#include <kernel/OS.h>
+#endif
+
+#ifdef PTHREAD_MAX_NAMELEN_NP
+#define MAX_THREAD_NAME_SIZE PTHREAD_MAX_NAMELEN_NP
+#elif defined(__APPLE__)
+#define MAX_THREAD_NAME_SIZE 63
+#elif defined(__FreeBSD__)
+#define MAX_THREAD_NAME_SIZE MAXCOMLEN
+#elif defined(__HAIKU__)
+#define MAX_THREAD_NAME_SIZE B_OS_NAME_LENGTH
+#else
+#define MAX_THREAD_NAME_SIZE 15
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Get the current thread ID.
+ *
+ * @return The current thread ID as a size_t value.
+ */
+static inline size_t minipal_get_current_thread_id(void)
+{
+#ifdef __wasm
+    return 0;
+#else
+    static __thread size_t tid = 0;
+
+    if (!tid)
+#if defined(__linux__)
+        tid = (size_t)syscall(SYS_gettid);
+#elif defined(__APPLE__)
+    {
+        uint64_t thread_id;
+        pthread_threadid_np(pthread_self(), &thread_id);
+        tid = (size_t)thread_id;  // Cast the uint64_t thread ID to size_t
+    }
+#elif defined(__FreeBSD__)
+        tid = (size_t)pthread_getthreadid_np();
+#elif defined(__NetBSD__)
+        tid = (size_t)_lwp_self();
+#else
+        tid = (size_t)(void*)pthread_self();
+#endif
+
+    return tid;
+#endif
+}
+
+/**
+ * Set the name of the specified thread.
+ *
+ * @param thread The thread for which to set the name.
+ * @param name The desired name for the thread.
+ * @return 0 on success, or an error code if the operation fails.
+ */
+static inline int minipal_set_thread_name(pthread_t thread, const char* name)
+{
+#ifdef __wasm
+    // WASM does not support pthread_setname_np yet: https://github.com/emscripten-core/emscripten/pull/18751
+    return 0;
+#else
+    const char* threadName = name;
+    if (strlen(name) > MAX_THREAD_NAME_SIZE)
+    {
+        char truncatedName[MAX_THREAD_NAME_SIZE + 1];
+        strncpy(truncatedName, name, MAX_THREAD_NAME_SIZE);
+        truncatedName[MAX_THREAD_NAME_SIZE] = '\0';
+        threadName = truncatedName;
+    }
```

Comment:
I'm not sure if this is definitely undefined behavior, but it feels bad to return a pointer to `truncatedName` outside  of its scope

Reply:
Good point. Shall move it outside.